numpy/lib/npyio.py

```diff
@@ -252,11 +238,17 @@ def __getitem__(self, key):
             magic = bytes.read(len(format.MAGIC_PREFIX))
             bytes.close()
             if magic == format.MAGIC_PREFIX:
-                bytes = self.zip.open(key)
-                return format.read_array(bytes,
-                                         allow_pickle=self.allow_pickle,
-                                         pickle_kwargs=self.pickle_kwargs,
-                                         max_header_size=self.max_header_size)
+                if self.mmap_mode:
+                    if self.allow_pickle:
+                        self.max_header_size = 2**64
+                    return format.open_memmap(self.zip.filename, mode=self.mmap_mode,
+                                              max_header_size=self.max_header_size, key=key)
```

Comment:
if this file contains more than one inputs, this code will become segmentfault:
x1 = np.arange((25), dtype=np.uint8)
x2 = np.arange((25), dtype=np.uint8)
b = {'x1': x1, 'x2':x2}